daemon/daemon_test.go

```diff
@@ -36,8 +36,10 @@ type daemonSuite struct{}
 var _ = check.Suite(&daemonSuite{})
 
 // build a new daemon, with only a little of Init(), suitable for the tests
-func newTestDaemon() *Daemon {
-	d := New()
+func newTestDaemon(c *check.C) *Daemon {
+	d, err := New()
+	c.Assert(err, check.IsNil)
+	c.Assert(d, check.Not(check.IsNil))
```

Comment:
Agree with @stevenwilkin, we don't check for the value not-nillness elsewhere after we've tested the err.